app/src/matchmaking/match/match.cpp

```diff
@@ -248,13 +248,9 @@ bool Match::playMove(Player& us, Player& them) {
 
     auto best_move = us.engine.bestmove();
 
-    if (best_move && !isUciMove(best_move.value())) {
-        setEngineIllegalMoveStatus(us, them, best_move, true);
-        return false;
-    }
-
-    const auto move  = best_move ? uci::uciToMove(board_, *best_move) : Move::NO_MOVE;
-    const auto legal = isLegal(move);
+    const auto move   = best_move ? uci::uciToMove(board_, *best_move) : Move::NO_MOVE;
+    const auto is_uci = isUciMove(best_move.value());
```

Comment:
This will throw an exception if no bestmove is found just use value_or("")

Reply:
alright, fixed 👍